src/main/java/org/gridsuite/network/map/dto/mapper/battery/AbstractBatteryInfos.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.network.map.dto.mapper.battery;
+
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.extensions.ActivePowerControl;
+import com.powsybl.iidm.network.extensions.ConnectablePosition;
+import com.powsybl.network.store.iidm.impl.MinMaxReactiveLimitsImpl;
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+import org.gridsuite.network.map.dto.ElementInfos;
+import org.gridsuite.network.map.dto.definition.battery.BatteryFormInfos;
+import org.gridsuite.network.map.dto.definition.battery.BatteryTabInfos;
+import org.gridsuite.network.map.model.MinMaxReactiveLimitsMapData;
+import org.gridsuite.network.map.model.ReactiveCapabilityCurveMapData;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.gridsuite.network.map.dto.utils.ElementUtils.nullIfNan;
+
+/**
+ * @author AJELLAL Ali <ali.ajellal@rte-france.com>
+ */
+
+@SuperBuilder
+@Getter
+public abstract class AbstractBatteryInfos extends ElementInfos {
+    public static ElementInfos toData(Identifiable<?> identifiable, InfoType dataType) {
+        switch (dataType) {
+            case TAB:
+                return toBatteryTabInfos(identifiable);
+            case FORM:
+                return toBatteryFormInfos(identifiable);
+            default:
+                throw new UnsupportedOperationException("TODO");
+        }
+    }
+
```

Comment:
public static ElementInfos toData(Identifiable<?> identifiable, ElementInfos.InfoType dataType) {
        return switch (dataType) {
            case TAB -> toBatteryTabInfos(identifiable);
            case FORM -> toBatteryFormInfos(identifiable);
            default -> throw new UnsupportedOperationException("TODO");
        };
    }

Reply:
this format is not supported by checkSyle